mempool/cat/pool.go

```diff
@@ -660,9 +660,14 @@ func (txmp *TxPool) recheckTransactions() {
 		"height", txmp.height,
 	)
 
+	// Get all transactions currently in the mempool requiring recheck.
+	// This avoids holding the store lock during the CheckTx calls which could
+	// cause a deadlock when handleRecheckResult tries to modify the store.
+	wtxs := txmp.store.getOrderedTxs()
+
 	// Issue CheckTx calls for each remaining transaction, and when all the
 	// rechecks are complete signal watchers that transactions may be available.
-	txmp.store.iterateOrderedTxs(func(wtx *wrappedTx) bool {
+	for _, wtx := range wtxs {
```

Comment:
this might fix the issue but I'm not sure 

does that make any sense @cmwaters ?

currently we're holding the mempool's mtx during recheck, then holding the store's read mtx when iterating, then holding the store's full mtx when deleting in `handleRecheckResult`

Reply:
> currently we're holding the mempool's mtx during recheck, then holding the store's read mtx when iterating, then holding the store's full mtx when deleting in handleRecheckResult

This sounds correct and I think the proposed solution works